storey/table.py

```diff
@@ -64,32 +97,42 @@ async def _get_or_load_static_attributes_by_key(self, key, attributes='*'):
                 self._set_static_attrs(key, {})
         return self._get_static_attrs(key)
 
-    def _set_aggregation_metadata(self, aggregates: List[FieldAggregator], use_windows_from_schema: bool = False):
-        self._use_windows_from_schema = use_windows_from_schema
-        self._aggregates = aggregates
-
-    async def _persist_key(self, key, event_data_to_persist):
-        key = get_hashed_key(key)
+    async def _internal_persist_key(self, key, event_data_to_persist):
         aggr_by_key = self._get_aggregations_attrs(key)
         additional_data_persist = self._get_static_attrs(key)
         if event_data_to_persist:
             if not additional_data_persist:
                 additional_data_persist = event_data_to_persist
             else:
                 additional_data_persist.update(event_data_to_persist)
-        await self._storage._save_key(self._container, self._table_path, key, aggr_by_key, self._partitioned_by_key,
-                                      additional_data_persist)
+        await self._storage._save_key(self._container, self._table_path, key, aggr_by_key,
+                                      self._partitioned_by_key, additional_data_persist)
+
+    def _set_aggregation_metadata(self, aggregates: List[FieldAggregator], use_windows_from_schema: bool = False):
+        self._use_windows_from_schema = use_windows_from_schema
+        self._aggregates = aggregates
+
+    def _init_flush_task(self):
+        if self._flush_task is None and self._flush_interval_secs > 0:
+            loop = asyncio._get_running_loop()
+            if loop:
```

Comment:
Having to check this probably means it gets called in too many places. It doesn't really make sense for `_init_flush_task` to be called from outside an async def.

Reply:
i call _init_flush_task from any flow that changes the cache (aggregations or static attrs)
the problem is that we allow setting static attrs on the table before we started running the loop (like in test_write_table_specific_columns) so in some cases _set_static_attrs() may be called before there's a running event loop

Reply:
So that's probably not the right place to call this method. One unexpected bad thing that could happen is there could be *another* event loop running. Best to keep this in the async context.

Reply:
why would there be another event loop?
_get_running_loop() only returns the running event loop if there's one or None

Reply:
I can definitely write code to do this. It's hard to predict when this will arise in the real world. For example, jupyter sometimes runs its own event loop, such that you would get it if you call get_running_loop().